src/components/shared/WithSize.js

```diff
@@ -37,16 +39,19 @@ export function withSize<
 > {
   return class WithSizeWrapper extends React.PureComponent<*, State> {
     _resizeListener: () => void;
+    _requestIdleTimeout = { timeout: 1000 };
     state = { width: 0, height: 0 };
 
     componentDidMount() {
       const container = findDOMNode(this); // eslint-disable-line react/no-find-dom-node
       if (!container) {
         throw new Error('Unable to find the DOMNode');
       }
-      this._resizeListener = () => {
-        this._updateWidth(container);
-      };
+      this._resizeListener = debounce(() => {
+        requestIdleCallbackPolyfill(() => {
+          this._updateWidth(container);
+        }, this._requestIdleTimeout);
+      }, 1000);
```

Comment:
Thanks!

Now the next thing to do is: use `debounce` and `requestIdleCallbackPolyfill` only if `document.hidden` is true; otherwise we want to call `_updateWidth` right away :)